TaskClass.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Enum representing possible task states
+
+public enum TaskStatus
+{
+	NotStarted,
+	InProgress,
+	Completed,
+	OnHold       // If some issue prevents progression on a task
+}
+
+// Enum representing priotity level of a task
+
+public enum TaskPriority
+{
+	Low,
+	Medium,
+	High,
+	Critical
+}
+
+public class Task
+{
+	// Properties for a task object
+
+	public string TaskName { get; private set; }
+	public string Description { get; private set; }
```

Comment:
Why is the the private set? Only the class will be able to change it's state?

Reply:
Habit from doing 1 filed exercises :D I removed all private setters